From 8cb90c2f51bc4ff9c1c6eb16949546c3640b5d00 Mon Sep 17 00:00:00 2001
From: Michael Jeanson <mjeanson@efficios.com>
Date: Fri, 14 Aug 2015 11:37:31 -0400
Subject: [PATCH 793/933] Fix: mm_page_alloc_extfrag instrumentation for kernel
 3.18.10

The following commit has been backported from kernel 3.19.2 into
3.18.10, introducing a change to mm_page_alloc_extfrag and therefore
provoking a build failure on a 3.18.10 kernel:

commit a4f3f96f850db69890f47984c60606a2362bee89
Author: Vlastimil Babka <vbabka@suse.cz>
Date:   Wed Feb 11 15:28:15 2015 -0800

    mm: when stealing freepages, also take pages created by splitting buddy page

    commit 99592d598eca62bdbbf62b59941c189176dfc614 upstream.

This patch adds the proper version check to allow building on 3.18.y.

Signed-off-by: Michael Jeanson <mjeanson@efficios.com>
Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 drivers/staging/lttng/instrumentation/events/lttng-module/kmem.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/staging/lttng/instrumentation/events/lttng-module/kmem.h b/drivers/staging/lttng/instrumentation/events/lttng-module/kmem.h
index 85c116b..85d13b9 100644
--- a/drivers/staging/lttng/instrumentation/events/lttng-module/kmem.h
+++ b/drivers/staging/lttng/instrumentation/events/lttng-module/kmem.h
@@ -202,6 +202,7 @@ LTTNG_TRACEPOINT_EVENT_INSTANCE_MAP(kmem_mm_page, mm_page_pcpu_drain,
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,19,2)	\
 	|| LTTNG_KERNEL_RANGE(3,14,36, 3,15,0)		\
+	|| LTTNG_KERNEL_RANGE(3,18,10, 3,19,0)		\
 	|| LTTNG_DEBIAN_KERNEL_RANGE(3,16,7,9,0,0, 3,17,0,0,0,0)	\
 	|| LTTNG_UBUNTU_KERNEL_RANGE(3,13,11,50, 3,14,0,0)	\
 	|| LTTNG_UBUNTU_KERNEL_RANGE(3,16,7,38, 3,17,0,0))
-- 
2.0.2

